go/pools/resource_pool.go

```diff
@@ -77,12 +83,18 @@ func NewResourcePool(factory Factory, capacity, maxCap int, idleTimeout time.Dur
 	rp := &ResourcePool{
 		resources:   make(chan resourceWrapper, maxCap),
 		factory:     factory,
+		available:   sync2.NewAtomicInt64(int64(capacity)),
 		capacity:    sync2.NewAtomicInt64(int64(capacity)),
 		idleTimeout: sync2.NewAtomicDuration(idleTimeout),
 	}
 	for i := 0; i < capacity; i++ {
 		rp.resources <- resourceWrapper{}
 	}
+
+	if idleTimeout != 0 {
+		rp.idleTimer = timer.NewTimer(idleTimeout / 10)
+		rp.idleTimer.Start(func() { rp.closeIdleResources() })
```

Comment:
This can just be `rp.closeIdleResources`

Reply:
Fixed